lib/dune

```diff
@@ -1,8 +1,8 @@
 (library
  (name STM)
  (public_name multicorecheck.stm)
- (modules STM)
- (libraries qcheck util))
+ (modules STM StmSpec STM_Seq STM_Common STM_Domain STM_Thread)
+ (libraries threads qcheck util))
```

Comment:
Here we need to express that `threads` is only a dependency of `STM_Thread` to avoid the implicit side-effects that the dependency causes (as mentioned in #66 for Lin).

Perhaps this can be done by simply splitting the `dune` rule?

Nit: module `STM` is gone, no?

Reply:
I've splited `STM` in several libraries so that `thread` is only a dependency of `STM_Thread`

The module `STM` is the point of entry and reexport `Spec` and the different functor.